File: cp_rust/src/bin/abc043-b-unhappy-hacking-abc-edit.rs
// Created by Ayush Biswas at 2026/01/20 20:34
// https://atcoder.jp/contests/abc043/tasks/abc043_b
use cp_lib::*;

// @code begin
use cpio::*;

sol! {
    fn solution(
        s: String
    ) -> String {
        s.chars().fold(vec![], |mut acc, c| {
            if c == 'B' {
                acc.pop();
            } else {
                acc.push(c)
            }
            acc
        }).into_iter().collect()
    }
}
// @code end

File: cp_haskell/bin/abc043-b-unhappy-hacking-abc-edit.hs
-- Created by Ayush Biswas at 2026/03/01 14:57
-- https://atcoder.jp/contests/abc043/tasks/abc043_b

-- @code begin

import Control.Monad.State (State, execState, get, modify)

transform :: Char -> String -> String
transform 'B' [] = []
transform 'B' (_ : xs) = xs
transform c xs = c : xs

keyb :: Char -> State String ()
keyb c = do modify (transform c)

main :: IO ()
main = do
  s <- getLine
  putStrLn . reverse $ execState (mapM_ keyb s) []

-- @code end